File: 1749/solution.go
package main

func maxAbsoluteSum(nums []int) int {
	n := len(nums)
	for i := 1; i < n; i++ {
		nums[i] += nums[i-1]
	}
	max := func(a, b int) int {
		if a > b {
			return a
		}
		return b
	}
	min := func(a, b int) int {
		if a < b {
			return a
		}
		return b
	}
	a, b := 0, 0
	for i := 0; i < n; i++ {
		a = max(a, nums[i])
		b = min(b, nums[i])
	}
	return a - b
}

File: 1749/solution.py
from typing import List


class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(1, n):
            nums[i] += nums[i - 1]
        a, b = 0, 0
        for i in range(n):
            a = max(a, nums[i])
            b = min(b, nums[i])
        return a - b